### DISCUSSION TITLE
Reading hdf4 data

### DISCUSSION THREAD

**Discussion by hvagasky** [ARM-DOE/pyart#1378] [Q&A]
Hi, 

I have NASA APR-2 data in an hdf4 file. I was hoping to use pyart.graph.AirborneRadarDisplay.plot_rhi to plot the data. However, pyart.io.read says that pyart hdf4 is an unsupported format. Is there a way to get around this issue? I can read the data into python using pyhdf. 

Thanks
Hannah

**Comment by scollis** [comment]
Hey Hannah,Here are the  un supported formats that will workhttps://arm-doe.github.io/pyart/API/generated/pyart.aux_io.htmlHere are the supported formatshttps://arm-doe.github.io/pyart/API/generated/pyart.io.html# You would need to ingest this into a radar object.  I would suggest starting a discussion on the discourse group where you may get more eyes on this https://openradar.discourse.group/ From: Hannah Vagasky ***@***.***>Date: Wednesday, January 18, 2023 at 10:35 AMTo: ARM-DOE/pyart ***@***.***>Cc: Subscribed ***@***.***>Subject: [ARM-DOE/pyart] Reading hdf4 data (Discussion #1378)Hi,I have NASA APR-2 data in an hdf4 file. I was hoping to use pyart.graph.AirborneRadarDisplay.plot_rhi to plot the data. However, pyart.io.read says that pyart hdf4 is an unsupported format. Is there a way to get around this issue? I can read the data into python using pyhdf.ThanksHannah—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by mgrover1** [comment]
@hvagasky I would suggest trying out the `ODIM_H5` reader, illustrated here
https://arm-doe.github.io/pyart/API/generated/pyart.aux_io.read_odim_h5.html#pyart.aux_io.read_odim_h5

```python
import pyart

radar = pyart.aux_io.read_odim_h5('your_file.h5')
```

You can also try using our new xradar package here
https://docs.openradarscience.org/projects/xradar/en/latest/notebooks/ODIM_H5.html

**Comment by kmuehlbauer** [comment]
@hvagasky Unfortunately the `ODIM_h5` reader proposed by @mgrover1 doesn't work for your hdf4 data.

You might be successful using rasterio/rioxarray and xarray:

```python
import xarray

xds = xarray.open_dataset(hdf4_filename, engine="rasterio") 
```

If that works, please show the output of 

```
# python interpreter
print(xds)
# jupyter lab or the like
display(xds)
```

to see, how this could be brought to Py-ART or otherwise make a decent plot.

**Reply by mgrover1** [reply]
Ope good point! Thanks @kmuehlbauer

**Reply by kmuehlbauer** [reply]
Yes, hdf4 is tricky, but good chance that this approach will work. I've tried with some MODIS hdf4 data I had lying around which imported nicely into xarray Dataset.

**Reply by kmuehlbauer** [reply]
Just adding relevant docs here:

https://corteva.github.io/rioxarray/stable/getting_started/getting_started.html
